### PR TITLE
Feature/jolokia plugin

### PR DESCRIPTION
**Pull request by saiello**
I wrote a simple JVM plugin based on jolokia rest api.

I needed to add some new methods in testutil/accumulator.go hence I refactor the oldest to use them.

### CODE REVIEW

**Comment by sparrc** [comment]
@saiello can you please write a README for this plugin?

you can use the CPU readme as an example: https://github.com/influxdb/telegraf/blob/master/plugins/system/CPU_README.md

**Comment by sparrc** [comment]
@saiello, I'm not super familiar with JMX/jolokia/MBeans, but it seems like this information is more generically called an `mbean`? Should the plugin be called `mbeans` or `JMX`?

File: plugins/jolokia/jolokia.go [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+	Servers []Server
+	Metrics []Metric
+	Tags    map[string]string
+}
+
+func (j *Jolokia) SampleConfig() string {
+	return `
+  context = "/jolokia/read"
+
+	[jolokia.tags]
```
Comment (sparrc) [inline plugins/jolokia/jolokia.go]: this should be space-indented

File: plugins/jolokia/jolokia.go [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+	jClient JolokiaClient
+	Context string
+	Servers []Server
+	Metrics []Metric
+	Tags    map[string]string
+}
+
+func (j *Jolokia) SampleConfig() string {
+	return `
+  context = "/jolokia/read"
```
Comment (sparrc) [inline plugins/jolokia/jolokia.go]: can you put some comments in here to briefly describe some of these options?

File: plugins/jolokia/jolokia.go [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+
+	[jolokia.tags]
+		group = "as"
```
Comment (sparrc) [inline plugins/jolokia/jolokia.go]: also space-indented

File: plugins/jolokia/jolokia.go [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+  [[jolokia.metrics]]
+        name = "memory_eden"
+        jmx  = "/java.lang:type=MemoryPool,name=PS Eden Space/Usage"
+        pass = ["used"]
+
+  [[jolokia.metrics]]
+        name = "heap_threads"
+        jmx  = "/java.lang:type=Threading"
+ #      drop = ["AllThread"]
```
Comment (sparrc) [inline plugins/jolokia/jolokia.go]: why is this line commented?

File: plugins/jolokia/jolokia.go [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+        pass = ["CurrentThreadCpuTime","CurrentThreadUserTime","DaemonThreadCount","ThreadCount","TotalStartedThreadCount"]
```
Comment (sparrc) [inline plugins/jolokia/jolokia.go]: line-length, split this list on multiple lines like:

```
pass = [
  "foo",
  "bar",
  ...
]
```

**Comment by sparrc** [comment]
@saiello nevermind about renaming the plugin, I get it now, `jolokia` is good :+1:

**Comment by sparrc** [comment]
Once you write a README and address my comments I'll merge it, thanks a bunch @saiello!

**Comment by saiello** [comment]
@sparrc I added a README to jolokia plugin. I hope to have fixed the indentation as well :grin: Thanks

**Comment by sparrc** [comment]
thank you @saiello!

**Pull request closed without merging**